packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -86,15 +90,93 @@ const VeniaAdapter = props => {
         return null;
     }
 
+    let storeCodeRouteHandler = '';
+    const browserRouterProps = {};
+    if (process.env.USE_STORE_CODE_IN_URL === 'true') {
+        const storeCode = storage.getItem('store_view_code') || STORE_VIEW_CODE;
+        browserRouterProps.basename = `/${storeCode}`;
+        const storeCodes = [];
+        const storeCurrencies = {};
+        if (Array.isArray(AVAILABLE_STORE_VIEWS)) {
+            AVAILABLE_STORE_VIEWS.forEach(store => {
+                storeCodes.push(store.code);
+                storeCurrencies[store.code] =
+                    store.default_display_currency_code;
+            });
+        }
+
+        /**
+         * The current store code won't be matched as it's included as the basename, if we match another store
+         * we need to change the current store.
+         */
+        storeCodeRouteHandler = (
+            <Switch>
+                <Route path={`/:storeCode(${storeCodes.join('|')})?`}>
+                    {({ match }) => {
+                        if (
+                            match.params.storeCode &&
+                            storeCodes.includes(match.params.storeCode)
+                        ) {
+                            /**
+                             * Only execute if one store code is present in the URL, multiple store codes will cause
+                             * the store state to break and cause weird side effects for the user
+                             */
+                            const regex = new RegExp(
+                                `(${storeCodes.join('|')})\/`,
+                                'g'
+                            );
+                            const storeCodesInUrl = window.location.pathname.match(
+                                regex
+                            ).length;
+                            if (storeCodesInUrl === 1) {
+                                storage.setItem(
+                                    'store_view_code',
+                                    match.params.storeCode
+                                );
+                                storage.setItem(
+                                    'store_view_currency',
+                                    storeCurrencies[match.params.storeCode]
+                                );
+                            } else {
+                                console.warn(
+                                    'Multiple store codes present in URL.'
```

Comment:
Should we/can we trim out extra store codes? Having multiple doesn't prevent us from rendering the page using the default (or first?) in the url.

Reply:
Multiple store codes in the URL causes a very interesting redirect loop. The regex match pattern for the `<Route />` will read the last match, rather than the first which is annoying.

As this isn't a valid use case, and people shouldn't be trying to navigate in this manner in an actual e-commerce store (it's incredibly rare for someone to go change the URL rather than use the switcher). I opted to just `warn` during development incase someone encounters this issue while debugging some weird use case they've got.  I also don't like doing URL rewrites as it gets even messier making sure we don't break the entire URL.